client/src/test/java/io/pravega/client/stream/impl/ClientFactoryTest.java

```diff
@@ -105,8 +106,10 @@ public void testEventWriterSealedStream() {
                 .thenReturn(CompletableFuture.completedFuture(currentSegments));
 
         EventWriterConfig writerConfig = EventWriterConfig.builder().build();
+        @Cleanup
         EventStreamWriter<String> writer = clientFactory.createEventWriter(stream, new JavaSerializer<String>(), writerConfig);
         assertEquals(writerConfig, writer.getConfig());
+        writer.close();
```

Comment:
Is this `close()` redundant to the `@Cleanup` annotation a few lines above?

Reply:
If the test fails somwhere before this line, the writer will never be cleaned up. So whatever connections and thread pools it opens (yes, it does that), will stick around. Cleanup is essentially rewriting the code as a `try-finally` to make sure that whatever we have declared as needing cleanup will be closed when the code exits